ivy/functional/ivy/activations.py

```diff
@@ -77,13 +79,35 @@ def gelu(
     return current_backend(x).gelu(x, approximate=approximate, out=out)
 
 
+@handle_device_shifting
+def _leaky_relu_jax_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    fn_original: Optional[Callable],
+    alpha: float = 0.2,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    return ivy.where(
+        (
+            ivy.logical_or(
+                ivy.real(x) < 0, ivy.logical_and(ivy.real(x) == 0, ivy.imag(x) < 0)
+            )
+        ),
+        x * alpha,
+        x,
+    )
+
+
 @handle_exceptions
 @handle_nestable
 @handle_array_like_without_promotion
 @handle_out_argument
 @to_native_arrays_and_back
 @handle_array_function
+@handle_complex_input
```

Comment:
Is there a reason why this decorator is above `handle_device_shifting`? 
If you put this decorator below it then you won't have to put `handle_device_shifting` decorator on `_leaky_relu_jax_like`

Reply:
Honestly, it was mostly because I couldn't exactly tell what `handle_device_shifting` does but it just felt more low-level than this one so I left it there. I was planning to ask someone where exactly in the stack this function is best placed, but I wanted to get it working well enough first